### ISSUE TITLE
The distro check looks bugged in Makefile

### ISSUE THREAD

**Issue by jiangzhengwenjz** [BluRosie/hg-engine#139]
Hello,

I'm on Fedora Linux. When building the tools,  make is using `csc` instead of `mcs`. I noticed that there's a switch `UBUNTU` for this, so I have to do `make build_tools --jobs UBUNTU=0`.

I believe this line (`UBUNTU = $(shell grep -i -q 'ubuntu' /proc/version; echo $$?)`) is not doing what  was expected. If I run the equivalent commands in my terminal:
```
$ grep -i -q 'ubuntu' /proc/version; echo $?
1
```
Clearly grep returns 1 when the string is not found.

**Comment by BluRosie** [comment]
may have to change it to default to use `msc` and switch based on debian presence instead of the other way around.  what do you see with `cat /proc/version`?  not sure how to detect distro effectively otherwise

**Comment by jiangzhengwenjz** [comment]
```
$ cat /proc/version
Linux version 6.5.5-100.fc37.x86_64 (mockbuild@bkernel02.iad2.fedoraproject.org) (gcc (GCC) 12.3.1 20230508 (Red Hat 12.3.1-1), GNU ld version 2.38-27.fc37) #1 SMP PREEMPT_DYNAMIC Sat Sep 23 22:53:27 UTC 2023
```

**Comment by BluRosie** [comment]
will probably just switch to `mcs` as default since having switched to mono from wsl anyway--would also simplify makefile for `msys2` and such, which is starting to look like i'm going to have to manually grab ndstool and armips executables from github releases

**Comment by BluRosie** [comment]
in this process i also got rid of devkitpro installation requirement, now just use arm-none-eabi things.  doesn't break existing devkitpro setups, but it's just no longer a requirement

also made it so that all of the tools have individual targets so they can build in parallel and such

am looking to test on macOS later today to make sure things don't break before closing this

6fb75f62f042a9e49057071306db898c6c718191

**Comment by BluRosie** [comment]
closing as done

**Comment by OrdinaryCloverOC** [comment]
idk if this is related, but: "requested access to resource is denied" when running the docker file "docker-makerom" the other linux native method gives error 127 upon running "make". (repeated error 127 if doing make --jobs (fedora)) should i open a new issue?
